core/domain/state_domain.py

```diff
@@ -110,11 +113,12 @@ def validate(self, interaction, exp_param_specs_dict):
                 'Expected answer group rules to be a list, received %s'
                 % self.rule_specs)
 
-        if self.tagged_misconception_id is not None:
-            if not isinstance(self.tagged_misconception_id, int):
+        if self.tagged_skill_misconception_id is not None:
+            if self.tagged_skill_misconception_id.find('-') == -1:
```

Comment:
You probably need to first check if tagged_skill_misconception_id is a string, otherwise this line will throw an error

Reply:
Done.